File: test_mercari_api.js
const https = require('https');

function testMercariApi(keyword) {
    const postData = JSON.stringify({
        userId: "",
        pageSize: 30,
        pageToken: "",
        searchSessionId: "test_session",
        indexRouting: "INDEX_ROUTING_UNSPECIFIED",
        thumbnailTypes: [],
        searchMode: "SEARCH_MODE_DEFAULT",
        customerType: "CUSTOMER_TYPE_UNSPECIFIED",
        keyword: keyword,
        status: ["STATUS_ON_SALE"],
        itemConditionId: [1, 2, 3],
        sort: "SORT_CREATED_TIME",
        order: "ORDER_DESC"
    });

    const options = {
        hostname: 'api.mercari.jp',
        port: 443,
        path: '/v2/entities:search',
        method: 'POST',
        headers: {
            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36',
            'Content-Type': 'application/json',
            'X-Platform': 'web',
            'Content-Length': Buffer.byteLength(postData)
        }
    };

    const req = https.request(options, (res) => {
        let data = '';
        res.on('data', chunk => data += chunk);
        res.on('end', () => {
            console.log('Mercari API Status:', res.statusCode);
            try {
                const parsed = JSON.parse(data);
                if (parsed.items) {
                    console.log(`✅ メルカリAPI取得成功! ${parsed.items.length} 件検出`);
                    console.log('先頭商品:', parsed.items[0].name, '¥' + parsed.items[0].price);
                } else {
                    console.log('API Response keys:', Object.keys(parsed));
                }
            } catch(e) {
                console.log('Response error:', e.message);
            }
        });
    });

    req.on('error', (e) => console.error('Mercari API Request Error:', e));
    req.write(postData);
    req.end();
}

testMercariApi('Nintendo Switch本体');

File: backup_v1_before_ui_unification/ebay-stock-monitor/backend/test_mercari_api.py
import httpx
import json

url = "https://api.mercari.jp/v2/entities:search"

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36",
    "X-Platform": "web",
    "Accept": "application/json, text/plain, */*",
    "Content-Type": "application/json"
}

payload = {
    "userId": "",
    "pageSize": 50,
    "pageToken": "",
    "searchSessionId": "test_session_12345",
    "indexRouting": "INDEX_ROUTING_UNSPECIFIED",
    "thumbnailTypes": [],
    "searchCondition": {
        "keyword": "Canon PowerShot S110",
        "excludeKeyword": "",
        "sort": "SORT_CREATED_TIME",
        "order": "ORDER_DESC",
        "status": ["STATUS_ON_SALE"],
        "sizeId": [],
        "categoryId": [],
        "brandId": [],
        "sellerId": [],
        "priceMin": 0,
        "priceMax": 46031,
        "itemConditionId": [],
        "shippingPayerId": []
    },
    "defaultDatasets": [],
    "serviceFrom": "suram"
}

resp = httpx.post(url, headers=headers, json=payload, timeout=10.0)
print("API Status:", resp.status_code)
if resp.status_code == 200:
    data = resp.json()
    items = data.get("items", [])
    print("API Items count:", len(items))
    for it in items[:10]:
        item_id = it.get("id")
        name = it.get("name")
        price = it.get("price")
        thumbnails = it.get("thumbnails", [])
        img = thumbnails[0] if thumbnails else ""
        print(f"[{price}円] {name} -> https://jp.mercari.com/item/{item_id}")
else:
    print("API Error Response:", resp.text[:300])
